src/v3/verify.test.ts

```diff
@@ -0,0 +1,145 @@
+import { verify } from "./verify";
+import sample from "./schema/wrapped-sample-document.json";
+import batched from "./schema/wrapped-batched-documents-1.json";
+
+import { cloneDeep } from "lodash";
+import { OpenAttestationVerifiableCredential, SignatureAlgorithm } from "../shared/@types/document";
+
+// sample1: unwrapped (aka credential), sample2: only 1 doc is wrapped (aka verifiable credential/VC)
+const sampleVerifiableCredential = sample as OpenAttestationVerifiableCredential;
+
+// sample3 & sample4: more than 1 doc wrapped (aka batched VC, where 'proofs' has values)
+const sampleBatched1 = batched as OpenAttestationVerifiableCredential;
+
+describe("signature", () => {
+  describe("verify", () => {
+    test("returns true for documents with unaltered data (without proofs)", () => {
+      // For this test case, proofs does not exist as sampleVerifiableCredential is wrapped on its own (i.e. targetHash == merkleRoot)
+      const verifiableCredential = sampleVerifiableCredential;
+      const verified = verify(verifiableCredential);
+      expect(verified).toBe(true);
+    });
+    test("returns false for documents with altered value", () => {
+      const verifiableCredential = {
+        ...cloneDeep(sampleVerifiableCredential),
+        recipient: {
+          name: "Fake Name"
+        }
+      };
+      const verified = verify(verifiableCredential);
+      expect(verified).toBe(false);
+    });
+    test("returns false for documents with altered key", () => {
+      const verifiableCredential = {
+        ...cloneDeep(sampleVerifiableCredential),
+        recipient: {
+          fakename: "Recipient Name"
+        }
+      };
+      const verified = verify(verifiableCredential);
+      expect(verified).toBe(false);
+    });
+    test("returns false for documents with added data", () => {
+      const verifiableCredential = {
+        ...cloneDeep(sampleVerifiableCredential),
+        credentialSubject: {
+          id: "did:example:SERIAL_NUMBER_123",
+          class: [
+            {
+              type: "3",
+              effectiveDate: "2010-01-01T19:23:24Z"
+            },
+            {
+              type: "3A",
+              effectiveDate: "2010-01-01T19:23:24Z"
+            },
+            {
+              type: "2A",
+              effectiveDate: "2020-06-05T00:00:00Z"
+            }
+          ]
+        }
+      };
+      const verified = verify(verifiableCredential);
+      expect(verified).toBe(false);
+    });
+
+    test("returns true for documents with unaltered data (with proofs)", () => {
```

Comment:
for all the (with proofs test), I would suggest to create a new describe block that will contain all of them

Reply:
updated!